src/Microsoft.Azure.SignalR/HubHost/ServiceLifetimeManagerCore.cs

```diff
@@ -0,0 +1,186 @@
+        {
+            if (IsInvalidArgument(groupName) || IsInvalidArgument(methodName))
+            {
+                return Task.CompletedTask;
+            }
+
+            var message = new GroupBroadcastDataMessage(groupName, excludedIds, SerializeAllProtocols(methodName, args));
+            // Send this message from a fixed service connection, so that message order can be reserved.
+            return _serviceConnectionContainer.WriteAsync(groupName, message);
+        }
+
+        public override Task SendUserAsync(string userId, string methodName, object[] args, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(userId) || IsInvalidArgument(methodName))
+            {
+                return Task.CompletedTask;
+            }
+
+            return _serviceConnectionContainer.WriteAsync(
+                new UserDataMessage(userId, SerializeAllProtocols(methodName, args)));
+        }
+
+        public override Task SendUsersAsync(IReadOnlyList<string> userIds, string methodName, object[] args,
+            CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(userIds) || IsInvalidArgument(methodName))
+            {
+                return Task.CompletedTask;
+            }
+
+            return _serviceConnectionContainer.WriteAsync(
+                new MultiUserDataMessage(userIds, SerializeAllProtocols(methodName, args)));
+        }
+
+        public override Task AddToGroupAsync(string connectionId, string groupName, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(connectionId) || IsInvalidArgument(groupName))
+            {
+                return Task.CompletedTask;
+            }
+
+            var message = new JoinGroupMessage(connectionId, groupName);
+            // Send this message from a fixed service connection, so that message order can be reserved.
+            return _serviceConnectionContainer.WriteAsync(groupName, message);
+        }
+
+        public override Task RemoveFromGroupAsync(string connectionId, string groupName, CancellationToken cancellationToken = default)
+        {
+            if (IsInvalidArgument(connectionId) || IsInvalidArgument(groupName))
+            {
+                return Task.CompletedTask;
+            }
+
+            var message = new LeaveGroupMessage(connectionId, groupName);
+            // Send this message from a fixed service connection, so that message order can be reserved.
+            return _serviceConnectionContainer.WriteAsync(groupName, message);
+        }
+
+        protected static bool IsInvalidArgument(string value)
+        {
+            return string.IsNullOrEmpty(value);
+        }
+
+        protected static bool IsInvalidArgument(IReadOnlyList<object> list)
```

Comment:
what about `new string[] { null }`?

Reply:
dotnet core signalr doesn't check it, but the current azure signalr sdk check it, may be we should keep the behavior the same as before?